core/src/main/java/io/jenkins/pluginhealth/scoring/probes/Probe.java

```diff
@@ -45,22 +45,47 @@ public final ProbeResult apply(Plugin plugin, ProbeContext context) {
     }
 
     /**
-     * Perform the analyze on a plugin
+     * Perform the analyze on a plugin.
+     * Based on the provided plugin and context, the method returns a non-null {@link ProbeResult}.
      *
      * @param plugin  the plugin on which the analyze is done
      * @param context holder of information passed across the probes executed on a single plugin
      * @return a ProbeResult representing the result of the analyze
      */
     protected abstract ProbeResult doApply(Plugin plugin, ProbeContext context);
 
+    /**
+     * Returns the key identifier for the probe.
+     * This is how the different probes can be identified in the {@link Plugin#details} map.
+     *
+     * @return the identifier of the probe
+     */
     public abstract String key();
 
+    /**
+     * Returns a description of the action of the probe.
+     *
+     * @return the description of the probe
+     */
     public abstract String getDescription();
 
+    /**
+     * Returns a boolean value which specify if the probe result can only be modified by a release of the plugin.
+     * If the probe execution is only altered by a new release of the plugin, returns true, otherwise returns false.
```

Comment:
```suggestion
     * Returns a boolean value that specifies if the probe result can only be modified by a release of the plugin.
     * If the probe result can be altered by a new release of the plugin, returns true, otherwise returns false.
```